src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,952 @@
+      We check after the refinement if the new value of the variable is outside its integer bounds and if that is the case, either revert to the old state or set it to bottom. *)
+
+  exception NotRefinable
+
+  let meet_tcons_one_var_eq res expr = res
+
+  (* meet_tcons -> meet with guard in if statement
+     texpr -> tree expr (right hand side of equality)
+     -> expression used to derive tcons -> used to check for overflow
+     tcons -> tree constraint (expression < 0)
+     -> does not have types (overflow is type dependent)
+  *)
+  let print_coeff_vec l (env : Environment.t) = 
+    let print_element _ e = match e with 
+      | (a, Some x) -> print_string ((Z.to_string a) ^ " * " ^ (Var.to_string ( Environment.var_of_dim env x)) ^ " + ") 
+      | (a, None) -> print_string ((Z.to_string a) ^ "+") in
+    List.fold_left print_element () l; print_newline ()
+
+  let print_final_expr l (env : Environment.t) =
+    let print_element _ i a = if i == 0 then print_string ((Z.to_string a) ^ " + ") else 
+        print_string ((Z.to_string a) ^ " * " ^ (Var.to_string ( Environment.var_of_dim env (i-1))) ^ " + ") in 
+    List.fold_lefti print_element () l; print_newline ()
+
+  let meet_tcons t tcons _ = 
+    (* The expression is evaluated using an array of coefficients. The first element of the array belongs to the constant followed by the coefficients of all variables 
+       depending on the result in the array after the evaluating including resolving the constraints in t.d the tcons can be evaluated and additional constraints can be added to t.d *)
+    let expr_init = Array.init ((Environment.size t.env) +1) (fun _ -> Z.zero) in 
+    match t.d with 
+    | None -> bot_env
+    | Some d -> if is_bot_env t then bot_env else
+        let cv's = get_coeff_vec t (Texpr1.to_expr @@ Tcons1.get_texpr1 tcons) in
+        let update (expr : Z.t Array.t)( c , v) = 
+          match v with 
+          | None -> Array.set expr 0 (Z.add expr.(0) c) ; expr 
+          | Some idx -> match d.(idx) with 
+            | (Some idx_i,c_i) -> Array.set expr 0 (Z.add expr.(0)  (Z.mul c  c_i)) ; Array.set expr (idx_i + 1) (Z.add expr.(idx_i + 1) c) ; expr
+            | (None, c_i) -> Array.set expr 0 (Z.add expr.(0)  (Z.mul c  c_i)) ; expr
+        in 
+        let final_expr = List.fold_left (fun expr cv -> update expr cv ) expr_init cv's in 
+        let var_count = List.count_matching (fun a -> if Z.equal a Z.zero then false else true) ( List.tl ( Array.to_list final_expr)) 
+        in 
+        if var_count == 0 then 
+          match Tcons1.get_typ tcons with 
+          | EQ -> if Z.equal final_expr.(0) Z.zero then t else bot_env
+          | SUPEQ -> if Z.geq final_expr.(0) Z.zero then t else bot_env
+          | SUP -> if Z.gt final_expr.(0) Z.zero then t else bot_env
+          | DISEQ ->  if Z.equal final_expr.(0) Z.zero then bot_env else t
+          | EQMOD scalar -> t (*Not supported right now
```

Comment:
Can you use `when`  clauses here, such that you can group all cases that return `t` and all cases that return `bot_env`?